src/melee.cpp

```diff
@@ -1870,11 +1870,11 @@ std::vector<special_attack> player::mutation_attacks(Creature &t) const
             }
 
             ///\EFFECT_UNARMED increases chance of attacking with mutated body parts
-
             ///\EFFECT_DEX increases chance of attacking with mutated body parts
-            const int chance = std::max( 1, mut_atk.chance - get_dex() - unarmed );
+            const float chance = get_dex() + unarmed / mut_atk.chance;
             add_msg( m_debug, "%s proc chance: %d", pr.first.c_str(), chance );
-            if( chance != 1 && !one_in( chance ) ) {
+            // If the mutation attack can fail to proc, and it does fail to proc, bail out
+            if( get_dex() + unarmed < mut_atk.chance && !x_in_y( get_dex() + unarmed, mut_atk.chance ) ) {
```

Comment:
I want to make it so you don't _need_ to make the chance = 1 for mutation attacks to occur more than half of the time. I'm not intending to change the values for mutation attacks to make them all unable to hit 100% proc chance with actually achievable stat/skill values at this time, and the attack chance can easily reach 100% for certain mutation lines e.g. Raptor muzzle.

Code quality is a good point. I suppose doing one calculation to see if you don't need to do a second near-identical calculation is kind of self-defeating.